### PR TITLE
Format all exceptions as reported by swagger

### LINKED ISSUE #57: Fix the typing of 500 response bodies.
When a 500 response is generated, it is supposed to be formatted according to the type shown for example in Swagger. 

To reproduce this error, generate a 500 by calling 
get /v1/taxonomy/main/concepts without search parameters.

The error message will look like this:
```
{
  "message": "clojure.lang.ExceptionInfo: [?c] not bound in expression clause: [(identity ?c) ?uniqueness] {:cognitect.anomalies/category :cognitect.anomalies/incorrect, :cognitect.anomalies/message \"[?c] not bound in expression clause: [(identity ?c) ?uniqueness]\", :db/error :db.error/insufficient-binding, :dbs [{:database-id \"0b3a02ce-799a-4b42-86f1-1cf06f5278e0\", :t 30, :next-t 31, :history false}]}"
}
```
and it's supposed to look like this:
```
{
  "taxonomy/error": "string"
}
```

### PR DESCRIPTION
**Pull request by vlaaad**
Closes #57

### CODE REVIEW

File: src/clj/jobtech_taxonomy_api/routes/services.clj [hunk 1]
```diff
@@ -1,46 +1,43 @@
...
+            [jobtech-taxonomy-api.db.core :as core]
+            [jobtech-taxonomy-api.db.daynotes :as daynotes]
+            [jobtech-taxonomy-api.webhooks :as webhooks]
+            [jobtech-taxonomy-api.store :as store]
+            [taxonomy :as types]
+            [clojure.tools.logging :as log]
+            [clojure.spec.alpha :as s]
+            [spec-tools.core :as st]
+            [spec-tools.data-spec :as ds]
+            [jobtech-taxonomy-api.routes.parameter-util :as pu]))
```
Comment (vlaaad) [inline src/clj/jobtech_taxonomy_api/routes/services.clj]: I use parinfer for structural editing, and it requires code to be properly formatted. Most changes in this file are caused by parinfer, but I'll highlight in the following comments what changes are relevant to the issue.

File: src/clj/jobtech_taxonomy_api/routes/services.clj [hunk 2]
```diff
@@ -56,32 +53,40 @@
...
       (resp/unauthorized (types/map->nsmap {:error "Not authorized"})))))
 
+(defn catch-errors [handler]
+  (fn [request]
+    (try
+      (handler request)
+      (catch Exception e
+        (log/error e "Error while performing request")
+        {:status 500
+         :body {::types/error (.getMessage e)}}))))
```
Comment (vlaaad) [inline src/clj/jobtech_taxonomy_api/routes/services.clj]: This is a middleware that properly reports all caught exceptions as "taxonomy/error" message.

Side note: `taxonomy` namespace is required and aliased as `types`, which puzzled me at first: why not just use `taxonomy` directly? It aligns better with what is returned in the API.

File: src/clj/jobtech_taxonomy_api/routes/services.clj [hunk 3]
```diff
@@ -56,32 +53,40 @@
...
+
 (defn service-routes []
   ["/v1/taxonomy"
    {:coercion spec-coercion/coercion
     :muuntaja formats/instance
+    :middleware [catch-errors]
```
Comment (vlaaad) [inline src/clj/jobtech_taxonomy_api/routes/services.clj]: Error catching middleware is added at the root of the router tree. Everything from now on is formatting changes.

File: src/clj/jobtech_taxonomy_api/routes/services.clj (hunk 1 above)
Comment (weipe) [reply src/clj/jobtech_taxonomy_api/routes/services.clj]: It's all good, I personally prefer trailing closing parathesis on the same line too.

File: src/clj/jobtech_taxonomy_api/routes/services.clj (hunk 2 above)
Comment (weipe) [reply src/clj/jobtech_taxonomy_api/routes/services.clj]: True! Feel free to refactor the code.

File: src/clj/jobtech_taxonomy_api/routes/services.clj (hunk 3 above)
Comment (weipe) [reply src/clj/jobtech_taxonomy_api/routes/services.clj]: This is excellent!

**Pull request merged**